docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+
+There's one last feature of `static using`. The `static using` directive
+also imports any nested types. That enables you to reference any nested
+types without qualification.
+
+## Null-conditional operators
+
+Null values complicate code. You need to check every access
+of variables to ensure you are not dereferencing `null`. The
+***null conditional operator*** makes those checks much easier
+and fluid.
+
+Simply replace the member access `.` with `?.`:
+
+[!code-csharp[NullConditional](../../samples/snippets/csharp/new-in-6/program.cs#NullConditional)]
+
+In the above example, the variable `first` is assigned `null` if the person object
+is null. Otherwise, it gets assigned the value of the `FirstName` property. Most importantly,
+the `?.` means that the above line of code does not generate a `NullReferenceException` when
+the `person` variable is null. Instead, it short-circuits and produces null.
+
+Also, note that this expression returns a `string`, regardless of the value of `person`.
+In the case of short circuiting, the null value returned is typed to match the full
+expression.
+
+You can often use this construct with the ***null coalescing*** operator to assign
+default values when one of the properties are null:
+
+[!code-csharp[NullCoalescing](../../samples/snippets/csharp/new-in-6/program.cs#NullCoalescing)]
+
+The right hand side operand of the `?.` operator is not limited to properties or fields.
+You can also use it to conditionally invoke methods. The most common use of member functions
+ with the null conditional operator is to safely invoke delegates
+(or event handlers) that may be null.  You'll do this by calling the delegate's `Invoke` method
+using the `?.` operator to access the member. You can see an example where we discuss 
+[delegate patterns](delegates-patterns.md#Handling-Null-Delegates).
+
+The rules of the `?.` operator ensure that the left hand side of the operator is
+evaluated only once. This is important and enables many idioms, including the
+example using event handlers. Let's start with the event handler usage. In previous
+versions of C#, you were encouraged to write code like this:
+
+```csharp
+var handler = this.SomethingHappened;
+if (handler != null)
+    handler(this, eventArgs);
+```
+
+This was preferred over a simpler syntax:
+
+```csharp
+// Not recommended
+// Introduces a race condition
+// The SomethingHappened event
+// may have subscribers when checked,
+// and those subscribers may have been
+// removed before the event is raised. 
+if (this.SomethingHappened != null)
+    this.SomethingHappened(this, eventArgs);
+```
+
+In this second version, the `SomethingHappened` event handler might
+be non-null when tested, but if other code removes a handler,
+it could still be null when the event handler was called.
+
+The compiler generates code for for the `?.` operator that ensures
+the left side (`this.SomethingHappened`) below is evaluated once, and the result
```

Comment:
style guide: don't use below.

also just a side comment here that we might want to file a bug:
I find it weird that the parentheses and the text inside get separated in the rendered view as this:
![image](https://cloud.githubusercontent.com/assets/12971179/18769313/ffa67cac-80e0-11e6-8fca-75fbc1be40c0.png)